openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -641,107 +641,108 @@ def password_change_request_handler(request):
         return HttpResponseBadRequest(_("No email address provided."))
 
 
-@require_POST
-@ensure_csrf_cookie
-def password_reset_token_validate(request):
-    """HTTP end-point to validate password reset token. """
-    is_valid = False
-    token = request.POST.get('token')
-    try:
-        token = token.split('-', 1)
-        uid_int = base36_to_int(token[0])
-        if request.user.is_authenticated and request.user.id != uid_int:
-            return JsonResponse({'is_valid': is_valid})
+class PasswordResetTokenValidation(APIView):
 
-        user = User.objects.get(id=uid_int)
-        if UserRetirementRequest.has_user_requested_retirement(user):
-            return JsonResponse({'is_valid': is_valid})
+    @method_decorator(ensure_csrf_cookie)
```

Comment:
Remove this decorator, this endpoint is meant to be available for anonymous users.